### PR TITLE
Parameterized diff_drive_controller's odom_frame_id

### PR DESCRIPTION
**Pull request by tappan-at-git**
Addressing issue #204, made odom_frame_id_ a parameter-driven value for diff_drive_controllers.

### CODE REVIEW

**Comment by bmagyar** [comment]
May I ask you to provide 2 tests as well?
One where you don't specify the topic, and check that it shows up on  `/odom`, and another one where you specify a different topic and the test checks that the new topic gets published and that there is no odom. Feel free to peek into the `test` folder for inspiration ;)

Updating the [diff drive wiki](http://wiki.ros.org/diff_drive_controller#Parameters) should also be in order.

**Comment by tappan-at-git** [comment]
Sorry, just so I'm sure I understand the tests you're asking for -- by topic do you mean frame id?

Currently working on:
- [x] Testing with `odom_frame_id` not specified: `/odom` tf exists, published odometry messages have their header frame_id set to `odom`.
- [x] Testing with `odom_frame_id` set to "new_odom": Original `/odom` tf does not exist, `/new_odom` tf does exist, and published odometry messages use `new_odom` in their header frame_id.

**Comment by tappan-at-git** [comment]
Tests implemented, and passing.

File: diff_drive_controller/test/diff_drive_default_odom_frame_test.cpp [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+TEST_F(DiffDriveControllerTest, testOdomTopic)
+{
+  // wait for ROS
+  while(!isControllerAlive())
+  {
+    ros::Duration(0.1).sleep();
+  }
+
+  // zero everything before test
```
Comment (efernandez) [inline diff_drive_controller/test/diff_drive_default_odom_frame_test.cpp]: optional: you don't need to zero anything for this test

File: diff_drive_controller/test/diff_drive_odom_frame.test [hunk 2]
```diff
@@ -0,0 +1,20 @@
+<launch>
+  <!-- Load common test stuff -->
+  <include file="$(find diff_drive_controller)/test/diff_drive_common.launch" />
+
+  <!-- Set odom_frame_id to something new -->
+  <rosparam>
+    diffbot_controller:
+      odom_frame_id: new_odom
+      enable_odom_tf: True
```
Comment (efernandez) [inline diff_drive_controller/test/diff_drive_odom_frame.test]: pedantic: `true`

File: diff_drive_controller/test/diff_drive_odom_frame_test.cpp [hunk 3]
```diff
@@ -0,0 +1,93 @@
+///////////////////////////////////////////////////////////////////////////////
+// Copyright (C) 2014, PAL Robotics S.L.
```
Comment (efernandez) [inline diff_drive_controller/test/diff_drive_odom_frame_test.cpp]: You can change the copyright owner and year (here and for the other test)

File: diff_drive_controller/test/diff_drive_odom_frame_test.cpp [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+TEST_F(DiffDriveControllerTest, testOdomTopic)
+{
+  // wait for ROS
+  while(!isControllerAlive())
+  {
+    ros::Duration(0.1).sleep();
+  }
+
+  // zero everything before test
```
Comment (efernandez) [inline diff_drive_controller/test/diff_drive_odom_frame_test.cpp]: ditto before

**Comment by efernandez** [comment]
Just some minor comments related with the test.

I think it's good to have the ability to set the odometry frame_id, as in `amcl`.
The implementation LGTM. Thanks for the tests; maybe it was enough with the second one.

:+1: @tappan-at-git Let us know if you either address my minor comments or not, to merge this.

**Comment by tappan-at-git** [comment]
Changes made!

**Comment by bmagyar** [comment]
:+1: here too

**Comment by efernandez** [comment]
Thanks! Let's merge this :tada:

**Comment by bmagyar** [comment]
Thanks for your contribution Eric!

**Comment by tappan-at-git** [comment]
No problem! :tada: Is there a good way to keep track of the release cycle so I can make sure the new feature gets documented on the wiki once it hits?

**Comment by bmagyar** [comment]
Not really. You can check back to see if there are new tags with release
version or take a look at the corresponding release discussion on the
issues list.

**Pull request merged**